src/main/kotlin/blackjack/domain/state/StateType.kt

```diff
@@ -0,0 +1,22 @@
+package blackjack.domain.state
+
+import blackjack.domain.Deck
+import blackjack.domain.Game
+
+enum class StateType(
+    private val condition: (Deck) -> Boolean,
+) {
+    BLACKJACK({ deck ->
+        deck.size == Game.INIT_TAKE_SIZE && deck.score() == Game.THRESHOLD
+    }),
+    BURST({ deck ->
+        deck.score() > Game.THRESHOLD
+    }),
+    STAY({ deck ->
+        deck.score() <= Game.THRESHOLD
+    });
+
+    companion object {
+        fun from(deck: Deck): StateType = values().find { it.condition.invoke(deck) } ?: STAY
```

Comment:
invoke 메서드는 지금처럼 별도로 메서드를 직접 명시해서 호출할 필요없이, 아래와 같이 호출할 수 있어요!😃
```suggestion
        fun from(deck: Deck): StateType = values().find { it.condition(deck) } ?: STAY
```